Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUAdapterImpl.cpp

```diff
@@ -144,7 +152,7 @@ AdapterImpl::~AdapterImpl()
     wgpuAdapterRelease(m_backing);
 }
 
-void AdapterImpl::requestDevice(const DeviceDescriptor& descriptor, CompletionHandler<void(Ref<Device>&&)>&& callback)
+void AdapterImpl::requestDevice(const DeviceDescriptor& descriptor, CompletionHandler<void(RefPtr<Device>&&)>&& callback)
```

Comment:
How confident are you that we shouldn't be returning an invalid device instead of nullptr?

Reply:
The specification says to reject the promise. So I don't think there is anywhere to return the invalid device.